docs/contracts.rst

```diff
@@ -959,9 +959,29 @@ virtual method lookup.
 
 Constructors
 ============
-A constructor is an optional function with the same name as the contract which is executed upon contract creation. 
+A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation. 
 Constructor functions can be either ``public`` or ``internal``.
 
+::
+
+    pragma solidity ^0.5.0;
+
+    contract A {
+        uint public a;
+
+        constructor(uint _a) internal {
+            a = _a;
+        }
+    }
+
+    contract B is A(1) {
+        constructor() public {}
+    }
+
+A constructor set as ``internal`` causes the contract to be marked as :ref:`abstract <abstract-contract>`.
+
+Prior to Version 0.5.0, constructors were defined as functions with the same name as the contract. This syntax is now deprecated.
```

Comment:
Should say `prior to version 0.4.21` if we merge this now. Also usually we put this kind of information into a `.. note::` block.